src/bench/descriptors.cpp

```diff
@@ -4,6 +4,7 @@
 
 #include <bench/bench.h>
 #include <key.h>
+#include <pubkey.h>
```

Comment:
`ECC_Start()` is defined in [key.h](https://github.com/bitcoin/bitcoin/blob/master/src/key.h#L185) so this include is unneeded.

Reply:
https://github.com/bitcoin/bitcoin/blob/f09d47b263459e27faf5416287255eb3aed369a2/src/pubkey.h#L335

Reply:
Thanks for pointing that out, that must be why the additional include.  Although, it doesn't appear to be needed.  Also, if you look at [verify script](https://github.com/bitcoin/bitcoin/blob/master/src/bench/verify_script.cpp#L21), it uses `EccVerifyHandle` but doesn't include `pubkey.h`.  It seems like maybe it's being included by another include as well.  If we add `pubkey.h` here, do you think it should also be added to `verify_script`?  I find the c/c++ include system really confusing and hard to trace where dependencies are being pulled in from.

Reply:
> Although, it doesn't appear to be needed.

In this project we follow our [Developer Notes](https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#source-code-organization):
> Every `.cpp` and `.h` file should `#include` every header file it directly uses classes, functions or other definitions from, even if those headers are already included indirectly through other headers.

---

> If we add `pubkey.h` here, do you think it should also be added to `verify_script`?

Not in this PR, to keep it focused.

---

> I find the c/c++ include system really confusing and hard to trace where dependencies are being pulled in from.

To make it easier we started to use [IWYU](https://github.com/include-what-you-use/include-what-you-use). See #24831 and other related PRs.

Reply:
> In this project we follow our [Developer Notes](https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#source-code-organization):

When I last read the developer-notes I must have overlooked this.  I agree and think it's a good practice to help with dependency confusion.  It would be better if the compiler enforced this behavior however because I've found more than a few places where this practice isn't implemented.  Perhaps I'll put in a PR for fixing other includes but I'm not sure if it's important enough to get others to review.

> To make it easier we started to use [IWYU](https://github.com/include-what-you-use/include-what-you-use). See https://github.com/bitcoin/bitcoin/pull/24831 and other related PRs.

Thanks, I'll look into it.